spatialmath/triangle.go

```diff
@@ -0,0 +1,201 @@
+package spatialmath
+
+import (
+	"math"
+
+	"github.com/golang/geo/r3"
+)
+
+// Triangle is three points and a normal vector.
+type Triangle struct {
+	p0 r3.Vector
+	p1 r3.Vector
+	p2 r3.Vector
+
+	normal r3.Vector
+}
+
+// NewTriangle creates a Triangle from three points. The Normal is computed; directionality is random.
+func NewTriangle(p0, p1, p2 r3.Vector) *Triangle {
+	return &Triangle{
+		p0:     p0,
+		p1:     p1,
+		p2:     p2,
+		normal: PlaneNormal(p0, p1, p2),
+	}
+}
+
+// ClosestPointTriangleCoplanarPoint takes a point, and returns the closest point on the triangle to the given point
+// The given point *MUST* be coplanar with the triangle. If it is known ahead of time that the point is coplanar, this is faster.
+func ClosestPointTriangleCoplanarPoint(t *Triangle, pt r3.Vector) r3.Vector {
```

Comment:
I didn't realize these triangle functions were currently unused (or at least I cant see any other references to them in this PR).  Can we just move them to the rest of coverage code?